x-pack/plugins/fleet/storybook/main.ts

```diff
@@ -5,16 +5,12 @@
  * 2.0.
  */
 
-import type { Configuration } from 'webpack';
-import { defaultConfig, WebpackConfig } from '@kbn/storybook';
+import { defaultConfigWebFinal } from '@kbn/storybook';
 
 module.exports = {
-  ...defaultConfig,
+  ...defaultConfigWebFinal,
   addons: ['@storybook/addon-essentials'],
   babel: () => ({
     presets: [require.resolve('@kbn/babel-preset/webpack_preset')],
   }),
-  webpackFinal: (config: Configuration) => {
-    return WebpackConfig({ config });
-  },
```

Comment:
webpackFinal usage in storybook main.ts somehow is causing issues with newly added dependency of `ts-node ` most likely an issue with storybook typescript setup 

https://github.com/storybookjs/storybook/issues/9610